addons/mail/models/mail_thread.py

```diff
@@ -3114,7 +3114,10 @@ def _notify_thread(self, message, msg_vals=False, **kwargs):
         recipients_data = self._notify_get_recipients(message, msg_vals, **kwargs)
         if not recipients_data:
             return recipients_data
-
+        # cache data fetched by manual query to avoid extra queries when reading user.partner_id
+        for r in filter(lambda r: r["uid"], recipients_data):
+            user = self.env["res.users"].browse(r["uid"])
+            self.env.cache.insert_missing(user, user._fields["partner_id"], [r["id"]])
```

Comment:
so this is mandatory? cannot we remove that LOC you mentioned which is not used anymore?

Reply:
Yes absolutely mandatory to cache the value when using `for_current_user=True` with `with_user(...)` (which is necessary for the fix) as it will access `self.env.user.partner_id` which is otherwise not in cache (it was already in cache for the non-`with_user` user, that's why the difference in query if we don't add it here).

Reply:
I realized it's used in many more places that just the line I mentioned which is no longer used.